netbsd/NetBSDProcessList.c

```diff
@@ -38,6 +39,22 @@ static long fscale;
 static int pageSize;
 static int pageSizeKB;
 
+static char const *freqSysctls[] = {
+#if defined(__powerpc__)
+   "machdep.intrepid.frequency.current",
+#endif
+#if defined(__mips__)
+   "machdep.loongson.frequency.current",
+#endif
+#if defined(__i386__) || defined(__x86_64__)
+   "machdep.est.frequency.current",
+   "machdep.powernow.frequency.current",
+#endif
```

Comment:
Any particular reason to exclude those on non-matching platforms?

Reply:
Basically a micro-optimization to avoid a couple of pointless system calls, otherwise the #ifdefs just provide context to what those variables are for.

Reply:
As those are only fallbacks for when the newer `sysctl`s can't be found, I don't see much value in keeping those. In particular as you might want to save that you skipped the newer entry point and thus also just could save the index of the `sysctl` which finally worked out.

Reply:
The code needs to be kept because the newer variables are only used on ARM currently. I'll tidy it up though.

Reply:
I didn't mean to "drop those `sysctl`s", but to drop the `#ifdef`s.

Another idea was to "remember" which you last used, so you won't have to search for it every time.